tensorflow_datasets/core/dataset_utils.py

```diff
@@ -178,7 +186,8 @@ def _graph_dataset_iterator(ds_iter, graph=None):
 
 
 @api_utils.disallow_positional_args(allowed=["dataset"])
-def as_numpy(dataset, graph=None):
+def as_numpy(dataset: tf.data.Dataset,
+             graph: Optional[tf.Graph] = None) -> Callable[..., Any]:
```

Comment:
This seems to be the wrong return annotations, as explained in the description. `as_numpy`. Could you use `@typing.overload` and `Nested` from my comment in https://github.com/tensorflow/datasets/pull/1554/files#r387846786
to return the correct type (`Nested[tf.Tensor]` -> `Nested[np.array]` and `Nested[tf.data.Dataset]` -> `Nested[Iterator[NestedDict[tf.Tensor]]]`)

Reply:
I have changed this code. Please take a look whether I understood correctly or not.